app/legacy/space-migration/space-migrator.ts

```diff
@@ -0,0 +1,78 @@
+import {spawn, ChildProcess} from "child_process"
+import fs from "fs-extra"
+import readline from "readline"
+
+// zdeps/brimcap migrate -zqd="run/data/spaces" -root="run/data/brimcap-root"
+export default class SpaceMigrator {
+  process: ChildProcess | null
+
+  constructor(readonly srcDir: string, readonly destDir: string) {}
+
+  needMigration() {
+    if (fs.existsSync(this.srcDir)) {
+      const files = fs.readdirSync(this.srcDir)
+      return files.some((f) => f.startsWith("sp_"))
```

Comment:
You should just have to check if this.srcDir exsits. When the brimcap migrate completes with no further spaces it deletes the `spaces` dir. IIRC the new lake is stored in a folder called "lake".